ktor-features/ktor-webjars/src/io/ktor/webjars/Webjars.kt

```diff
@@ -0,0 +1,85 @@
+package io.ktor.webjars
+
+import io.ktor.application.ApplicationCall
+import io.ktor.application.ApplicationCallPipeline
+import io.ktor.application.ApplicationFeature
+import io.ktor.application.call
+import io.ktor.http.ContentType
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.defaultForFilePath
+import io.ktor.pipeline.PipelineContext
+import io.ktor.request.httpMethod
+import io.ktor.request.uri
+import io.ktor.response.respond
+import io.ktor.response.respondBytes
+import io.ktor.util.AttributeKey
+import org.webjars.MultipleMatchesException
+import org.webjars.WebJarAssetLocator
+import java.nio.file.Paths
+
+class Webjars(val configuration: Configuration) {
+
+    private fun fileName(path: String) : String = Paths.get(path).fileName.toString()
+
+    private fun extractWebJar(path : String) : String {
+        val firstSlash = if (path.startsWith("/")) 1 else 0
+        val nextSlash = path.indexOf("/", 1)
+        val webjar = if(nextSlash > -1) path.substring(firstSlash,  nextSlash) else ""
+        val partialPath = path.substring(nextSlash + 1)
+        return locator.getFullPath(webjar, partialPath)
+    }
+
+    private val locator = WebJarAssetLocator()
+
+    class Configuration {
+        var path = "/webjars"
+        set(value) {
+            var buffer = StringBuilder(value)
+            if(!value.startsWith("/")){
+                buffer.insert(0, "/")
+            }
+            if(value.endsWith("/")){
+                buffer.deleteCharAt(buffer.length - 1)
+            }
+            field = buffer.toString()
+        }
+    }
+
+    private suspend fun intercept(context: PipelineContext<Unit, ApplicationCall>){
+        val fullPath = context.call.request.uri
+        if(fullPath.startsWith(configuration.path) && context.call.request.httpMethod == HttpMethod.Get){
+            val fileName = fileName(context.call.request.uri)
+            val partialPath = fullPath.replace(configuration.path, "")
+            try {
+                val location = extractWebJar(partialPath)
+                context.call.respondBytes(ContentType.defaultForFilePath(fileName), HttpStatusCode.OK) {
```

Comment:
This will not provide versions tag so Condition headers support will not work that is VERY important for static files. You can try `JarFileContent` instead or provide etag based on webjar's version/checksum.

Reply:
It was my understanding that the fat jar would explode all jars inside the fat jar. This is why I did not use a `JarFileContent` as it would not find the jar inside the fat jar. Using `getResourceAsStream` would at least give me a inputStream to the contents. I need to figure out a way to associate a version now